Lib/http/server.py

```diff
@@ -144,6 +146,12 @@ def server_bind(self):
 class ThreadingHTTPServer(socketserver.ThreadingMixIn, HTTPServer):
     daemon_threads = True
 
+    def server_close(self):
+        # close TCP socket and more if override into HTTPServer class
+        HTTPServer.server_close(self)
+        # join all process request subthread
+        socketserver.ThreadingMixIn.server_close(self)
```

Comment:
I do not understand why this change is needed. `ThreadingMixIn.server_close()` calls `super().server_close()`, which in this case should be `TCPServer.server_close()`, and `HTTPServer.server_close()` is the same as `TCPServer.server_close()`. Perhaps this change is outdated?

Reply:
Hello, that's an old subject, I found the code that led me to make this PR. I recall myself the subjet and come back to you with more explanation

Reply:
ok I get it, you are right, the important part in this PR is the `super().shutdown()` in `TCPServer` (I.e. the change in socketserver.py).

Imho this change in `ThreadingHTTPServer` is only for convenience. If someone extends `ThreadingHTTPServer` it should only call `super.server_close()` and this should delegate to both parent class (but I am not a python expert maybe I miss something about mixin)

And in this case the newly `super.server_close()` will call `HTTPServer.server_close(self)` which is as you said `TCPServer.server_close()` and so with the seconds change will call `super().shutdown()` which is `BaseServer.shutdown(self)`


ps :  a toy project https://antoine-roux.tk/projects/python/bpo-40305 to reproduce and play with different case.

Reply:
I try to add test for this code too

Reply:
Ok right I remove this code it was indeed usefull